Autocomplete extensions (allow duplicates)

dlukez opened the pull request:
* add support for overwriting display-property in the auto-complete
* add support for specifying a custom track-property on the tags-input (allow duplicate labels). the autocomplete will listen to this too if set.

Closes #265
Closes #257
Closes #152
Closes #145

dmregister commented:
@mbenford,

Any chance that you can merge @dlukez changes? This would be a great feature to add to this module. I am will to help if any changes need to be made for the merge.

mbenford commented:
@dmregister I'll have some spare time in the following days and I'll try to review/merge as many PRs as I can.

dmregister commented:
@mbenford great! Thanks again for the module!

bastien-tilkee commented:
+1, I wouldd need this on a project as well

mbenford commented:
@dlukez Could you rebase this PR on top of the master branch?

dlukez commented:
Looking at rebasing this now...

dlukez commented:
Rebased and added some stuff in the test page to make sure it still works. Seems to be okay - someone else may want to verify this if they are using it in a current project?

mbenford commented on src/tags-input.js:
```diff
@@ -10,6 +10,7 @@
  *
  * @param {string} ngModel Assignable angular expression to data-bind to.
  * @param {string=} [displayProperty=text] Property to be rendered as the tag label.
+ * @param {string=} [trackProperty=text] Property to be used for tracking items in the ng-repeat.
```
I think `keyProperty` would be more appropriate. It's an implementation detail that it's used by `ng-repeat` to track items, and the user doesn't need to know that. The same is true for the option's description.

akhil10x5 commented:
Hello, I am currently using ngTagsInput version 3.0.0. I just wanted to know what option to be used to allow duplicates.
P.S: i went through the documentation page and could not figure out what to use.
